src/directives/infinite-scroll/infinite-scroll.component.spec.ts

```diff
@@ -0,0 +1,127 @@
+import { async, ComponentFixture, TestBed } from '@angular/core/testing';
+import { InfiniteScrollModule } from './infinite-scroll.module';
+import { Component } from '@angular/core';
+import { FormsModule } from '@angular/forms';
+
+
+@Component({
+    template: `<div class="row">
+                    <div class="col-md-6 col-sm-12">
+                        <input id="filter" type="text" class="form-control" placeholder="Filter"
+                            [(ngModel)]="filterText">
+                    </div>
+                </div>
+                <div class="row m-t-sm">
+                    <div class="col-md-6 col-sm-12">
+
+                        <div class="employee-list"
+                            [uxInfiniteScroll]="load"
+                            [filter]="filterText"
+                            [pageSize]="pageSize"
+                            [loadOnScroll]="loadOnScroll">
+
+                        </div>
+
+                    </div>
+                </div>
+    `
+})
+export class InfiniteScrollTestComponent {
+
+    filterText: any;
+    pageSize = 20;
+    loadOnScroll: boolean = false;
+    loading: boolean = false;
+    exhausted: boolean = false;
+
+    load(pageNum: number, pageSize: number, filter: any): any[] {
+        return [];
+    }
+
+}
+
+describe('Directive - Infinite Scroll', () => {
+    let component: InfiniteScrollTestComponent;
+    let fixture: ComponentFixture<InfiniteScrollTestComponent>;
+    let loadSpy: jasmine.Spy;
+
+    beforeEach(async(() => {
+        TestBed.configureTestingModule({
+            imports: [InfiniteScrollModule, FormsModule],
+            declarations: [InfiniteScrollTestComponent],
+        })
+            .compileComponents();
+    }));
+
+    beforeEach(() => {
+        fixture = TestBed.createComponent(InfiniteScrollTestComponent);
+        component = fixture.componentInstance;
+        loadSpy = spyOn(component, 'load').and.callThrough();
+        component.filterText = undefined;
+        fixture.detectChanges();
+    });
+
+    it ('should initialise correctly', () => {
```

Comment:
We can probably remove this test as none of the tests will pass if the fixture doesn't get initialized